### PR TITLE
feat!: add support for the `u256` type

### LINKED ISSUE #1797: Add support for the `u256` type
### Motivation

`u256` is one of the native Sway number types. We currently don't have support this type.

The Rust SDK does support it.

### PR DESCRIPTION
**Pull request by Dhaiwat10**
This PR adds missing support for the numerical primitive Sway type `u256`.

- [x] Coder (v0 and v1)
- [x] Typegen
- [x] Docs

## **❗️Breaking Changes:** 
- The `U64Coder` has been _removed_ in favour of `BigNumberCoder` (`abi-coder` package). This new coder contains the encoding/decoding logic for both u64 and u256 since both of their coders had almost exactly the same logic.

Closes #1797

### CODE REVIEW

File: apps/docs/src/guide/types/numbers.md [hunk 1]
```diff
@@ -1,42 +1,40 @@
...
 
-> Note: The TS SDK does not have support for the `u256` type yet.
-
 ## Creating Numbers
 
-### For `u64`
+### For `u64` and `u256`
 
-When you pass in a `u64` to a Sway program from JavaScript, you must first convert it to a `BigNum` object. Here's how you can do that:
+When you pass in a `u64` or a `u256` to a Sway program from JavaScript, you must first convert it to a `BigNum` object. Here's how you can do that:
```
Comment (danielbate) [inline apps/docs/src/guide/types/numbers.md:17]: Maybe add some reasoning here as to why?

File: packages/abi-coder/src/encoding/coders/v0/U256Coder.ts
Comment (danielbate) [inline packages/abi-coder/src/encoding/coders/v0/U256Coder.ts:1]: IMO the `U64Coder` and `U256Coder` could share the same coder, with some conditional logic to set the correct encoded length and slicing? Much alike the `NumberCoder` sharing `u8`, `u16` etc.

**Review by danielbate — CHANGES_REQUESTED** [review]
Great work Dhai 💪🏻 

To validate the `v1` changes, please could you add a `sway` test inside the `forc-projects-experimental` workspace and `experimental-logging.test.ts` to test a `v1` log.

File: packages/abi-coder/src/encoding/coders/v0/U256Coder.ts
Comment (arboleya) [reply packages/abi-coder/src/encoding/coders/v0/U256Coder.ts:1]: Good catch!

**Review by arboleya — DISMISSED** [review]
Very nice! 💯 

I second @danielbate comments.

File: packages/abi-coder/src/encoding/coders/v0/U256Coder.ts
Comment (Dhaiwat10) [reply packages/abi-coder/src/encoding/coders/v0/U256Coder.ts:1]: Yeah good point. I did think of this but purely in the interest of speed I didn't add it in this PR. But I'm gonna do it now!

**Review by Torres-ssf — DISMISSED** [review]

File: .changeset/silly-hats-end.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+"@fuel-ts/abi-coder": patch
+"@fuel-ts/abi-typegen": patch
```
Comment (arboleya) [inline .changeset/silly-hats-end.md:3]: Should it be a minor that it became a breaking change?
```suggestion
"@fuel-ts/abi-coder": minor
"@fuel-ts/abi-typegen": minor
```

File: .changeset/silly-hats-end.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+---
+
+feat: add missing support for the `u256` type
+
+The TS SDK is now capable of handling `u256` types, and hence capable of interacting (encoding/decoding) with any Sway programs that use the `u256` type.
```
Comment (arboleya) [inline .changeset/silly-hats-end.md:8]: We could also have a "breaking change" section with the changeset so it'll make it to the changelog.

**Review by arboleya — CHANGES_REQUESTED** [review]

**Review by danielbate — APPROVED** [review]

File: .changeset/silly-hats-end.md (hunk 2 above)
Comment (danielbate) [reply .changeset/silly-hats-end.md:3]: ➕

**Review by danielbate — APPROVED** [review]

**Review by nedsalk — APPROVED** [review]

**Pull request merged**